TestIntegrationExecCmdEnforcesTimeoutFlag fails: cluster error string lacks SQLSTATE 57014 prefix

\`make test-integration\` fails on \`origin/main\` (currently at \`9e6e8c1\`) in \`cmd\`:

\`\`\`
--- FAIL: TestIntegrationExecCmdEnforcesTimeoutFlag (1.61s)
    exec_integration_test.go:124:
        Error: "query execution canceled due to statement timeout" does not contain "57014"
        Test: TestIntegrationExecCmdEnforcesTimeoutFlag
        Messages: --timeout=1ms must produce a SQLSTATE 57014 error from the cluster
\`\`\`

The test asserts the error envelope's message string contains the substring \`"57014"\`. The actual message returned by the cluster is just the prose \`"query execution canceled due to statement timeout"\` with no SQLSTATE prefix in the message body. SQLSTATE 57014 is presumably still attached to the underlying \`pgerror\`, but the assertion is looking in the wrong place.

Reproduces against a CockroachDB \`v24.3.1\` cluster started via \`docker run cockroachdb/cockroach:latest start-single-node --insecure\`.

**Scope**
- Pick the right assertion target: either inspect the structured envelope's \`code\` field rather than the message string, or include SQLSTATE in the rendered message.
- Out of scope: the safety / timeout behaviour itself works — only the test's assertion is wrong.

**Demo**
\`make test-integration\` passes against a running CRDB cluster (\`COCKROACH_BIN\` set or \`CRDB_TEST_DSN\` configured).

**Files**
- \`cmd/exec_integration_test.go\`
- possibly \`cmd/exec.go\` if message rendering needs to include SQLSTATE

Surfaced while running \`/dev-loop\` for the README update PR (#135) — pre-existing on main, unrelated to that PR.

## Dependencies
- **Depends on:** none
- **Blocks:** none

--- a/cmd/exec_integration_test.go
+++ b/cmd/exec_integration_test.go
@@ -121,7 +121,7 @@ func TestIntegrationExecCmdEnforcesTimeoutFlag(t *testing.T) {
 	var env output.Envelope
 	require.NoError(t, json.Unmarshal(stdout.Bytes(), &env))
 	require.Len(t, env.Errors, 1)
-	require.Contains(t, env.Errors[0].Message, "57014",
+	require.Equal(t, "57014", env.Errors[0].Code,
 		"--timeout=1ms must produce a SQLSTATE 57014 error from the cluster")
 	require.Equal(t, output.ConnectionDisconnected, env.ConnectionStatus,
 		"a cluster-side timeout error leaves the manager's recovery contract in place — connection_status reverts")

